components/mediation/tasks/org.wso2.micro.integrator.mediation.ntask/src/main/java/org/wso2/micro/integrator/mediation/ntask/NTaskTaskManager.java

```diff
@@ -227,7 +223,10 @@ public boolean pauseAll() {
                 }
                 List<TaskInfo> taskList = taskManager.getAllTasks();
                 for (TaskInfo taskInfo : taskList) {
-                    taskManager.pauseTask(taskInfo.getName());
+                    String taskName = taskInfo.getName();
+                    if (!taskManager.isDeactivated(taskName)) {
```

Comment:
Do we need this validation here, it seems like a validation that needs to happen inside the task manager

Reply:
moved to task manager .